### PR TITLE
[Enhancement] Decouple components and create new factories

### PR DESCRIPTION
**Pull request by macrigiuseppe**
Signed-off-by: Giuseppe Macri <gmacri@uber.com>

### CODE REVIEW

File: src/constants/default-settings.js [hunk 1]
```diff
@@ -694,3 +694,6 @@ export const MAP_CONFIG_DESCRIPTION = 'Map config will be included in the Json f
 export const BASE_SPEED = 600;
 export const DEFAULT_TIME_FORMAT = 'MM/DD/YY HH:mm:ssa';
 export const SPEED_CONTROL_RANGE = [0, 10];
+
+// No Op
+export const NO_OP = () => {};
```
Comment (heshan0131) [inline src/constants/default-settings.js]: do we really need to add a N0_OP to default settings? feel like it's a little overkill, adding one line of import every file to use no_op probably not that pretty

File: src/components/modal-container.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ import DataTableModalFactory from './modals/data-table-modal';
 import LoadDataModalFactory from './modals/load-data-modal';
 import ExportImageModalFactory from './modals/export-image-modal';
 import ExportDataModalFactory from './modals/export-data-modal';
-import ExportMapModalFactory from './modals/export-map-modal';
+import ExportMapModalFactory from './modals/export-map-modal/export-map-modal';
```
Comment (heshan0131) [inline src/components/modal-container.js]: you should also export `ExportMapModalFactory` from `components/index.js`, so people can replace it with custom one

**Comment by jonsadka** [comment]
This is awesome 🙏

File: src/constants/default-settings.js (hunk 1 above)
Comment (macrigiuseppe) [reply src/constants/default-settings.js]: this was removed

**Pull request merged**